src/main/java/seedu/address/model/person/AttendanceList.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Represents a Student's Attendance.
+ */
+public class AttendanceList {
+    private List<Attendance> attendanceList;
+
+    /**
+     * Represents an Attendance record for a person.
+     * Each Attendance object stores whether the person has attended and the date of attendance.
+     *
+     * @param hasAttended A Boolean indicating if the person has attended.
+     * @param date The date and time of the attendance.
+     * @throws NullPointerException if {@code hasAttended} or {@code date} is null.
+     */
+    public AttendanceList() {
+        this.attendanceList = new ArrayList<Attendance>();
+    }
+
+    /**
+     * Adds an attendance record to the attendance list.
+     *
+     * @param attendance The attendance record to be added. Must not be null.
+     */
+    public void addAttendance(Attendance attendance) {
+        requireNonNull(attendance);
+        attendanceList.add(attendance);
+    }
+
+    /**
+     * Removes the specified attendance from the attendance list.
+     *
+     * @param attendance The attendance to be removed. Must not be null.
+     * @throws NullPointerException if the specified attendance is null.
+     */
+    public void removeAttendance(Attendance attendance) throws NullPointerException {
+        requireNonNull(attendance);
+        attendanceList.remove(attendance);
+    }
```

Comment:
Sorry, I meant a custom exception like the one in `UniquePersonList.java`, the `remove` method. But it's alright, we will just fix this in another PR.